src/json.hpp.re2c

```diff
@@ -7892,6 +7867,159 @@ class basic_json
         return result;
     }
 
+
+    /*!
+    @brief locale-independent serialization for built-in arithmetic types
+    */
+    struct numtostr
+    {
+    public:
+        template<typename T>
+        numtostr(T value) 
+        {
+            x_write(value, std::is_integral<T>());
+        }
+
+        operator const char*() const
+        {
+            return m_buf.data();
+        }
+
+        const char* c_str() const
+        {
+            return m_buf.data();
+        }
+
+    private:
+        static constexpr size_t s_capacity = 30;
+        std::array<char, s_capacity + 2> m_buf{{}}; // +2 for leading '-'
+                                                    // and trailing '\0'
+        template<typename T>
+        void x_write(T x, std::true_type)
+        {
+            static_assert(std::numeric_limits<T>::digits10 <= s_capacity, "");
+
+            const bool is_neg = x < 0;
+            size_t i = 0;
+
+            while(x and i < s_capacity) 
+            {
+                const auto digit = std::labs(static_cast<long>(x % 10));
+                m_buf[i++] = static_cast<char>('0' + digit);
+                x /= 10;
+            }
+
+            assert(i < s_capacity);
+
+            if(i == 0) 
+            {
+                m_buf[i++] = '0';
+            }
+
+            if(is_neg) 
+            {
+                m_buf[i++] = '-';
+            }
+
+            std::reverse(m_buf.begin(), m_buf.begin() + i);
+        }
+
+        template<typename T>
+        void x_write(T x, std::false_type)
+        {
+            if (x == 0)
+            {
+                std::strcpy(m_buf.data(), 
+                            std::signbit(x) ? "-0.0" : "0.0");
+                return;
+            }
+
+            static constexpr auto d = 
+                    std::numeric_limits<number_float_t>::digits10;
+            static_assert(d == 6 or d == 15 or d == 16 or d == 17, "");
+
+            static constexpr auto fmt = d == 6  ? "%.7g"
+                                      : d == 15 ? "%.16g"
+                                      : d == 16 ? "%.17g"
+                                      : d == 17 ? "%.18g"
+                                      :           "%.19g";
+            // I'm not sure why we need to +1 the precision,
+            // but without it there's a unit-test that fails
+            // that asserts precision of the output
+
+            snprintf(m_buf.data(), m_buf.size(), fmt, x);
+
+#if 0
```

Comment:
Yes - this is a here-be-dragons note to a future clever contributor (or code peruser) who'll think "let's use c++-flavor locales here!" Feel free to get rid of it.